examples/zkevm-demo/core/src/lib.rs

```diff
@@ -94,31 +91,36 @@ impl Database for ZkDb {
 pub mod ether_trace {
     use std::sync::Arc;
 
-    use bytes::Bytes;
-    use ethers_core::types::{BlockId, Transaction, H160, U64};
+    use ethers_core::types::{BlockId, Transaction, H160 as EthersH160, U256 as EthersU256, U64};
     use ethers_providers::Middleware;
     pub use ethers_providers::{Http, Provider};
-    use revm::primitives::{CreateScheme, TransactTo, TxEnv};
+    use revm::primitives::{Bytes, CreateScheme, TransactTo, TxEnv};
     use tokio::runtime::Handle;
 
     use super::*;
 
+    pub fn from_ethers_u256(x: EthersU256) -> U256 {
+        U256::from_limbs(x.0)
+    }
+
     pub fn txenv_from_tx(tx: Transaction) -> TxEnv {
         TxEnv {
             caller: tx.from.0.into(),
             gas_limit: tx.gas.as_u64(),
-            gas_price: tx.gas_price.unwrap().into(),
-            gas_priority_fee: tx.max_priority_fee_per_gas.map(|x| x.into()),
+            gas_price: from_ethers_u256(tx.gas_price.unwrap()),
+            gas_priority_fee: tx.max_priority_fee_per_gas.map(|x| from_ethers_u256(x)),
             transact_to: if let Some(to_addr) = tx.to {
                 TransactTo::Call(to_addr.0.into())
             } else {
                 TransactTo::Create(CreateScheme::Create) // TODO: create2
             },
-            value: tx.value.into(),
+            value: from_ethers_u256(tx.value),
             data: Bytes::from(tx.input.to_vec()), // TODO: gotta be a cleaner way
             chain_id: tx.chain_id.map(|elm| elm.as_u64()),
             nonce: Some(tx.nonce.as_u64()),
-            access_list: Vec::new(), // TODO
+            access_list: Vec::new(),
```

Comment:
Not having the access list of the transaction will lead to the EVM computing the wrong gas cost. The way how this example is designed, the gas cost is not used anywhere, so I am not sure whether it should be added especially since the conversion is kinda ugly:
```rust
  .map(|item| {
    (
      item.address.0.into(),
      item.storage_keys
        .into_iter()
        .map(|key| U256::from_be_bytes(key.into())).collect(),
      )
  }).collect()
```